generate-stats

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import time
+import sqlite3
+import argparse
+import logging
+import json
+
+# Seconds in two weeks
+TWO_WEEKS = 1209600
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate statistics about failed tests')
+    parser.add_argument("--db", default="cockpit-failures.db", help="Database name")
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose logging")
+    opts = parser.parse_args()
+
+    if opts.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    db_conn = sqlite3.connect(opts.db)
+    cursor = db_conn.cursor()
+
+    all_tables = []
+
+    # Get all failures sorted by number of occurrences
+    top10 = []
+    rows = cursor.execute("""\
+            SELECT testname, COUNT(run)
+            FROM Failures
+            GROUP BY testname
```

Comment:
Let's restrict this to "last two weeks" as well, for consistency? Or was this deliberate for some reason? If so, please point that out in the headings.

Reply:
Oh right, that was just forgotten. Fixed.